snowpark_style_guide.md

```diff
@@ -0,0 +1,329 @@
+# Snowpark Style Guide
+
+The Snowpark library provides intuitive APIs for querying and processing data in a data pipeline.
+Using this library, you can build applications that process data in Snowflake without having to move data to the system where your application code runs.
+
+This guide to Snowpark code style demonstrates common situations we've encountered and the associated best practices based on our experience coaching and working with our engineering teams.
+
+Beyond Snowpark specifics, the general practices of clean code are important in Snowpark -- the
+[PEP 8][pep-8] is a strong starting point for learning more about these practices.
+
+# Table of Contents
+
+1. [DataFrame Operations](#dataframe-operations)
+   1. [Column Selection](#column-selection)
+   2. [Aliasing Column Name](#aliasing-column-name)
+   3. [Performant SQL Generation](#performant-sql-generation)
+2. [Expressions](#expressions)
+   1. [Multiple-Lines](#multiple-lines)
+3. [Explicitness](#explicitness)
+   1. [Calling Methods and Functions with Parameter Names](#calling-methods-and-functions-with-parameter-names)
+4. [User Defined Functions (UDFs)](#user-defined-functions-udfs)
+   1. [Registration](#registration)
+   2. [Type Hints](#type-hints)
+5. [User Defined Table Functions (UDTFs)](#user-defined-table-functions-udfts)
```

Comment:
This is probably typo: `udfts` -> `udtfs`

Reply:
nice catch, updated